## Практическое задание № 5 ##

### Урок 5. Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование


1. Известно, что генеральная совокупность распределена нормально
со средним квадратическим отклонением, равным 16.
Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95,
если выборочная средняя M = 80, а объем выборки n = 256.

2. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью,
получены опытные данные:
6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1
Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,
оценить истинное значение величины X при помощи доверительного интервала, покрывающего это
значение с доверительной вероятностью 0,95.


Решать через тест гипотезы

3. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

4. Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

In [6]:
import math
import numpy as np

#### Задача N 1
Известно, что генеральная совокупность распределена нормально
со средним квадратическим отклонением, равным 16.
Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95,
если выборочная средняя M = 80, а объем выборки n = 256.

In [7]:
M = 80
n = 256
za2 = 1.96
sigma = 16
i1 = M-za2*sigma/math.sqrt(n)
i2 = M+za2*sigma/math.sqrt(n)
(i1, i2)

(78.04, 81.96)

> Доверительный интервал равен [78.04; 81.96]

#### Задача N 2
В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью,
получены опытные данные:
6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1
Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,
оценить истинное значение величины X при помощи доверительного интервала, покрывающего это
значение с доверительной вероятностью 0,95.

In [12]:
n=10
v=n-1
ta2 = 2.262
vb = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
M = vb.mean()
sigma = vb.std(ddof=1)
i1 = M-ta2*sigma/math.sqrt(n)
i2 = M+ta2*sigma/math.sqrt(n)
(i1, i2)

(6.267538255912426, 6.912461744087575)

> Доверительный интервал равен [6.268; 6913]

#### Задача N 3
Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

H1: m > m0

In [14]:
n = 100
M = 17
mu = 17.5
sigma = 2 / np.sqrt(100)
zt = 1.65
zn = (mu - M) / sigma
(zt, zn)

(1.65, 2.5)

> гипотеза H1 - верна. Утверждение неверно.

#### Задача N 4
Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

H1: m != m0